src/pip/_internal/vcs/git.py

```diff
@@ -163,16 +163,21 @@ def obtain(self, dest):
 
     def get_url(self, location):
         """Return URL of the first remote encountered."""
-        remotes = self.run_command(
-            ['config', '--get-regexp', r'remote\..*\.url'],
-            show_stdout=False, cwd=location)
-        remotes = remotes.splitlines()
-        found_remote = remotes[0]
-        for remote in remotes:
-            if remote.startswith('remote.origin.url '):
-                found_remote = remote
-                break
-        url = found_remote.split(' ')[1]
+        try:
+            remotes = self.run_command(
+                ['config', '--get-regexp', r'remote\..*\.url'],
+                show_stdout=False, cwd=location)
+            remotes = remotes.splitlines()
+            found_remote = remotes[0]
+            for remote in remotes:
+                if remote.startswith('remote.origin.url '):
+                    found_remote = remote
+                    break
+            url = found_remote.split(' ')[1]
+        except InstallationError:
```

Comment:
It seems like this try-except should be more tightly bound -- like just around `run_command()`.

Reply:
Also, one thing I'm wondering about is if this PR can possibly mask other issues, since the absence of remotes isn't the only thing that can cause an `InstallationError` (non-zero exit status). I'm wondering if there might be a tighter condition than catching all non-zero exit statuses.

Reply:
Ya we can more tightly scope the error checking. As for masking errors, I'm not sure if git config will return a variety of error codes we can check. But even if it does would it be invalid to report a local path in that case?